pkg/providers/instancetype/instancetypes.go

```diff
@@ -140,7 +144,7 @@ func (p *Provider) createOfferings(sku *skewer.SKU, zones sets.Set[string]) []cl
 }
 
 // getInstanceTypes retrieves all instance types from skewer using some opinionated filters
-func (p *Provider) getInstanceTypes(ctx context.Context) (map[string]*skewer.SKU, error) {
+func (p *Provider) getInstanceTypes(ctx context.Context, imageFamily string) (map[string]*skewer.SKU, error) {
 	if cached, ok := p.cache.Get(InstanceTypesCacheKey); ok {
```

Comment:
I am a little concerned about making getInstanceTypes parameterized/conditional on imageFamily here. It will also break the caching (which is currently not aware of potential differences based on imageFamily); will need to either use imageFamily as part of the key, or - better - do any imageFamily-dependent filtering later.

Reply:
Ah very fair callout. Should image family dependent filtering happen inside of the instance provider then? We should probably cache all instances, then filter out in launch instance based on image family?

Reply:
It can still be in the instance types provider, just after getInstanceTypes. (Yet another cache for instance types with offerings is missing there - right now we are processing SKUs every time! _That_ cache should be keyed smarter, including off image family.)

Reply:
yeah I agree about excessive iteration, we can filter out there too. For now just did that to keep the changes here minimal, we can create a followup issue to target that piece of work.